src/main/java/picard/fingerprint/CrosscheckFingerprints.java

```diff
@@ -139,12 +147,29 @@
     private double[][] crosscheckMatrix = null;
     private final List<String> matrixKeys = new ArrayList<>();
 
+
+
     @Override
     protected int doWork() {
         // Check inputs
-        INPUT.forEach(IOUtil::assertFileIsReadable);
+        IOUtil.assertFilesAreReadable(INPUT);
+        IOUtil.assertFilesAreReadable(INPUT2);
+
         IOUtil.assertFileIsReadable(HAPLOTYPE_MAP);
         if (OUTPUT != null) IOUtil.assertFileIsWritable(OUTPUT);
+
+        if (!INPUT2.isEmpty()) {
+            log.warn("INPUT2 is not empty. NOT doing cross-check. Will only compare each SAMPLE in INPUT1 against that sample in INPUT2");
+            if (MATRIX_OUTPUT != null) {
```

Comment:
Can this be done with a MUTEX for the argument?

Reply:
done.